### ISSUE TITLE
kube-dns restarts frequently，using kubeadm v1.6.0-alpha.0.2074

### ISSUE THREAD

**Issue by fangqiusheng** [kubernetes/kubeadm#99]
Seems that kube-dns is extremely unstable ,always CrashLoopBackOff, what could be the problem?

**Comment by pires** [comment]
Possible related to https://github.com/kubernetes/kubeadm/issues/102.

**Comment by tristanz** [comment]
We saw this on on earlier release (have not tested 1.6) and it seemed to be related to liveness check failure due to low CPU limits causing timeouts.

**Comment by fangqiusheng** [comment]
Problem is liveness check failure, forgot to set  net/bridge/bridge-nf-call-iptables=1, which must be set for cni network , details in http://kubernetes.io/docs/admin/network-plugins/#network-plugin-requirements

**Comment by pires** [comment]
@dmmcquay you're the preflight checks man. Can you take care of this?

**Comment by dmmcquay** [comment]
@pires and @luxas I'll take a look at this one.

**Comment by clearbjli** [comment]
@fangqiusheng What Linux distribution?

**Comment by bjornl** [comment]
@fangqiusheng What Linux distribution?

**Comment by fangqiusheng** [comment]
CentOS 7.2



qiusheng fang
mail：fangqiusheng@unionpay.com
 
发件人： Björn
发送时间： 2017-01-09 23:46
收件人： kubernetes/kubeadm
抄送： Fang Qiusheng; Mention
主题： Re: [kubernetes/kubeadm] kube-dns restarts frequently，using kubeadm v1.6.0-alpha.0.2074 (#99)
@fangqiusheng What Linux distribution?

**Comment by bjornl** [comment]
@fangqiusheng If it is kube-dns that can not communicate to the kube-apiserver I did experience the same problem on CentOS 7.2. I isolated it to the iptables INPUT chain was blocking traffic from pod network to the kube network. I suspect firewalld config by kubeadm may be brokenon CentOS.

If it is the case one can see the API connect failure in the log.

kubectl logs kube-dns-2924299975-pn60c --namespace=kube-system --container=kube-dns
